### PR TITLE
Look up atomic locations in the right place

### PR DESCRIPTION
**Pull request by jonhoo**

### CODE REVIEW

File: src/rt/atomic.rs [hunk 1]
```diff
@@ -535,7 +531,7 @@ impl State {
         if let Some(loaded) = current.ahead(&self.unsync_loaded_at) {
             location::panic("Causality violation: Concurrent `unsync_load` and mut accesses.")
                 .location("created", self.created_location)
-                .thread("mut", loaded, self.unsync_loaded_locations[loaded])
+                .thread("with_mut", loaded, self.unsync_mut_locations[loaded])
```
Comment (carllerche) [inline src/rt/atomic.rs:534]: I think this tests concurrent `unsync_load`, so the two conflicts should be "unsync load 1" and "unsync load 2"?

Comment (jonhoo) [reply src/rt/atomic.rs:534]: Hmm, why would two `unsync_load`s race? I wonder if it's the `ahead` check that's wrong, and should be checking `unsync_mut_at`?

**Comment by jonhoo** [comment]
As far as I can tell, it basically isn't possible to write a test for dc7eb04ce63791475de3167169f8ad6aa71a1c64 without some really unsafe code, since `with_mut` requires `&mut self`, so you wouldn't be able to call `unsync_load` at the same time without some raw pointer magic. I could write that though I suppose.

**Comment by jonhoo** [comment]
Do we _also_ want a test for concurrent unsynced loads (the way that check was originally written)? I don't see how that'd be a race (it's read-read), but that _was_ what it did...

**Pull request merged**